test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs

```diff
@@ -55,39 +54,30 @@ public void RunMultipleTestAssembliesWithoutTestAdapterPath(RunnerInfo runnerInf
         [TestMethod]
         [NetFullTargetFrameworkDataSource]
         [NetCoreTargetFrameworkDataSource]
-        [DoNotParallelize]
         public void RunMultipleTestAssembliesInParallel(RunnerInfo runnerInfo)
         {
             AcceptanceTestBase.SetTestEnvironment(this.testEnvironment, runnerInfo);
-            var resultsDir = GetResultsDirectory();
+            using var workspace = new Workspace();
 
             var assemblyPaths = this.BuildMultipleAssemblyPath("SimpleTestProject.dll", "SimpleTestProject2.dll").Trim('\"');
-            var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), string.Empty, this.FrameworkArgValue, runnerInfo.InIsolationValue, resultsDirectory: resultsDir);
+            var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), string.Empty, this.FrameworkArgValue, runnerInfo.InIsolationValue, resultsDirectory: workspace.Path);
             arguments = string.Concat(arguments, " /Parallel");
             arguments = string.Concat(arguments, " /Platform:x86");
-            string testhostProcessName = string.Empty;
-            // for the desktop we will run testhost.x86 in two copies, but for core 
-            // we will run a combination of testhost.x86 and dotnet, where the dotnet will be 
+            arguments += GetDiagArg(workspace.Path);
+
+            // for the desktop we will run testhost.x86 in two copies, but for core
+            // we will run a combination of testhost.x86 and dotnet, where the dotnet will be
             // the test console, and sometimes it will be the test host (e.g dotnet, dotnet, testhost.x86, or dotnet, testhost.x86, testhost.x86)
             // based on the target framework
-            int expectedNumOfProcessCreated = this.IsDesktopRunner() ? 2 : 3;
-            var testhostProcessNames = new[] { "testhost.x86", "dotnet" };
-
-            var cts = new CancellationTokenSource();
-            var numOfProcessCreatedTask = NumberOfProcessLaunchedUtility.NumberOfProcessCreated(
-                cts,
-                testhostProcessNames);
+            int expectedNumOfProcessCreated =
+                this.testEnvironment.TargetFramework == IntegrationTestBase.CoreRunnerFramework ? 1 : 2;
```

Comment:
This seems odd. Why would the .net framework have more processes running? I would expect to have the same number as we have dlls, or less for the .net framework, but not the other way around.

Reply:
If you look at the logs in standard output in Test Explorer you will see that it outputs two .host. files every time. You are then checking the line to match testhost and dotnet, but the line is testhost.exe or testhost.dll or testhost.x86.dll, so the core runner is not matching because you are trying to match `dotnet`. 

Looking at how this is used, there is potential false positive in the cases where `testhost` is matched because that is a substring of `testhost.x86`. The `Contains` check needs to be improved.